### ISSUE TITLE
[1.1.17] Text rendering order on overlapping text is wrong (again)

### ISSUE THREAD

**Issue by kristofbolyai** [RaphiMC/ImmediatelyFast#91]
The issue is basically the same as https://github.com/RaphiMC/ImmediatelyFast/issues/50. Looks like 1.1.17 undid the fixes. I can confirm that 1.1.16 worked.

If you need any specific info, logs, let me know.

**Comment by RaphiMC** [comment]
Can you provide screenshots and a way to reproduce it please

**Comment by kristofbolyai** [comment]
![image](https://github.com/RaphiMC/ImmediatelyFast/assets/49001742/1e1dc439-22af-4387-8130-3058af126da3)

Combined with `Wynntils Artemis v0.0.3-alpha.213`. This happens with all versions of the mod, but I would recommend testing against that version, as it is the most stable.

As for reproducing, it seems that this happens less frequently than it did before 1.1.16. The root of the issue is the same, rendering 2 fonts on top of each other, and also rendering some semi-transparent texture triggers it. In Wynntils' sense, this is the low health indicator and the spell cast renderer vignette, which both render semi-transparent textures.

**Comment by kristofbolyai** [comment]
This is how the right side is supposed to look like for reference.
![image](https://github.com/RaphiMC/ImmediatelyFast/assets/49001742/7ad42691-cfe7-4158-afcd-d1c0a41e0b24)

**Comment by SaltSouls** [comment]
I'm also, having this issue, but it is with the level and potion text. Would you be willing to send your mod list so that I could possibly compare and figure out if there is a common mod we are using that could be causing this?

**Comment by RaphiMC** [comment]
@SaltSouls Does it happen with only IF for you or not? Can you try that please

**Comment by SaltSouls** [comment]
It seems to be a bug with the newest version... after downgrading from 1.2.3 to 1.2.1, the issue went away.
I will try 1.2.3 by itself to see if it still happens.

**Comment by RaphiMC** [comment]
Are you using Forge or Fabric?

**Comment by SaltSouls** [comment]
Forge.

**Comment by RaphiMC** [comment]
The Forge version had broken HUD batching (It didn't do anything) until 1.2.2. So I assume its the HUD batching causing issues on forge

**Comment by SaltSouls** [comment]
Yeah, the issue persists even when IM is the only mod installed. I will add that this seems to be a 1.19 issue as when testing on 1.20 this problem doesn't appear.

**Comment by RaphiMC** [comment]
@SaltSouls The xp level text issue will be fixed in the next release (#115)

**Comment by kurtramone** [comment]
I can confirm this issue with last version of ImmediatlyFast
Maybe layers out of order?
MC 1.19.2
Forge: 43.2.21
IM: 1.2.3

Some exemples:
From Inventory no problems:
![Captura de tela 2023-09-12 220246](https://github.com/RaphiMC/ImmediatelyFast/assets/55619656/43974039-250d-469b-a5c1-f08bf9c12e2c)
From backpack missing info
![Captura de tela 2023-09-12 220300](https://github.com/RaphiMC/ImmediatelyFast/assets/55619656/560d7621-a05d-41d6-a83a-a0ec99367cd3)
XP text, but you guys already now
![Captura de tela 2023-09-12 220347](https://github.com/RaphiMC/ImmediatelyFast/assets/55619656/ab83f648-2e6d-4e2f-b46b-fbef8782bacf)
From OneProbe seems some layers are out of order
![Captura de tela 2023-09-12 220523](https://github.com/RaphiMC/ImmediatelyFast/assets/55619656/e0bff56f-9750-4e6c-956e-fc215efcf1ec)
Had some issues with AE2 too, but I forgot to save

**Comment by RaphiMC** [comment]
@kurtramone Can you try with this version and see how many issues remain: https://upload.raphimc.net/file/NSEEbZs9dlTy3cYu/Pf3mvZZuLy7z42ex/ImmediatelyFast-1.2.4-SNAPSHOT%2B1.19.2.jar

**Comment by kurtramone** [comment]
> @kurtramone Can you try with this version and see how many issues remain: https://upload.raphimc.net/file/NSEEbZs9dlTy3cYu/Pf3mvZZuLy7z42ex/ImmediatelyFast-1.2.4-SNAPSHOT%2B1.19.2.jar

Got a crash with this file
[crash-2023-09-13_14.14.05-client.txt](https://github.com/RaphiMC/ImmediatelyFast/files/12600250/crash-2023-09-13_14.14.05-client.txt)
It this file for Forge?

**Comment by SaltSouls** [comment]
Seems it caused some kinda issue with computer craft.

**Comment by RaphiMC** [comment]
Fixed the issue, please try again:
https://upload.raphimc.net/file/00skGjeBm8hhOe5D/kFomHd9oXyxZbx7n/ImmediatelyFast-1.2.4-SNAPSHOT%2B1.19.2.jar
https://upload.raphimc.net/file/ojkmbC8AtNq8Xnjn/zoJStRrFUuf7B3zw/ImmediatelyFast-1.2.4-SNAPSHOT%2B1.20.1.jar

**Comment by kurtramone** [comment]
> Fixed the issue, please try again: https://upload.raphimc.net/file/00skGjeBm8hhOe5D/kFomHd9oXyxZbx7n/ImmediatelyFast-1.2.4-SNAPSHOT%2B1.19.2.jar https://upload.raphimc.net/file/ojkmbC8AtNq8Xnjn/zoJStRrFUuf7B3zw/ImmediatelyFast-1.2.4-SNAPSHOT%2B1.20.1.jar

This snapshot works, no crashes, just little issues with UI, still looks out of order/transparent/fade

OneProbe with issues
![Captura de tela 2023-09-13 171634](https://github.com/RaphiMC/ImmediatelyFast/assets/55619656/5550bb39-fc8d-4338-83ec-98aecc66aceb)
Stats in inventory OK
![Captura de tela 2023-09-13 171651](https://github.com/RaphiMC/ImmediatelyFast/assets/55619656/2fd240e0-54d3-430e-a76e-89f29de27c3f)
Stats from back pack invisble
![Captura de tela 2023-09-13 171711](https://github.com/RaphiMC/ImmediatelyFast/assets/55619656/27d38ea6-b5d0-4966-b88a-8708436b1945)
XP text OK
![Captura de tela 2023-09-13 171740](https://github.com/RaphiMC/ImmediatelyFast/assets/55619656/b78ff047-0e31-4a95-8f69-e84e4290f542)
OneProbe from energy cube (Mekanism), it can't load energy bar at bottom/blank artifact
![Captura de tela 2023-09-13 171819](https://github.com/RaphiMC/ImmediatelyFast/assets/55619656/ab2a3134-dfc0-4baa-9a00-100850dc5eac)



You can see what I mean by looking at this older version
IF 1.2.1
![Captura de tela 2023-09-13 172543](https://github.com/RaphiMC/ImmediatelyFast/assets/55619656/58d0163c-a66e-414b-b546-2836cf632041)
![Captura de tela 2023-09-13 172450](https://github.com/RaphiMC/ImmediatelyFast/assets/55619656/9a8c623a-5488-4d9b-8b1b-0886930d5b7f)
![Captura de tela 2023-09-13 172517](https://github.com/RaphiMC/ImmediatelyFast/assets/55619656/cdcada1e-a3d9-4e89-87b5-a244bd6781f8)
![Captura de tela 2023-09-13 172555](https://github.com/RaphiMC/ImmediatelyFast/assets/55619656/4a699c32-7081-416d-9df3-93947c53cd0d)
![Captura de tela 2023-09-13 172359](https://github.com/RaphiMC/ImmediatelyFast/assets/55619656/03823890-fb74-475d-be15-179e6d0f13c0)

I'll keep testing this snapshot version for more issues.

**Comment by RaphiMC** [comment]
The UI issues with modded elements will be fixed at a later point. Its a bit more complex to fix those properly

**Comment by kurtramone** [comment]
Thanks for helping, that was super fast, if you need help with 1.19.2 just let me know (I'll wait for NeoForge in 1.20.2 to test it).

**Comment by RaphiMC** [comment]
Is this issue still relevant with the latest version of ImmediatelyFast?

**Comment by kurtramone** [comment]
Hello. I don't have 1.19 pack anymore. Sorry about that. I have since build a new PC, and I'm still using your mod on 1.21.1, but haven't noticed issues.

**Comment by RaphiMC** [comment]
I am going to close this then and if anyone still experiences this bug, please open a new issue
